main.cpp

```diff
@@ -18,13 +18,27 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
-public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    Rewriter& rewrite;
+ public:
+    CastCallBack(Rewriter& rewriter) rewrite(rewriter) {};
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        const auto *expr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        if (expr != nullptr) {
+            auto loc = CharSourceRange::getCharRange(
+                expr->getLParenLoc(), expr->getSubExprAsWritten()->getBeginLoc());
+            auto tSource = Lexer::getSourceText(
+                CharSourceRange::getTokenRange(

```

Comment:
Please descrbe the difference between the use of `getCharRange()` and `getTokenRange()` with examples.

Reply:
`getCharRange()` returns range that specifies characters with the last character included by the specified locations, while `getTokenRange()` returns range from the start of the range up to the start of the last token in the given range.

This difference can be seen by the following code:
```
  auto charRange = Lexer::getAsCharRange(
      CharSourceRange::getCharRange(node->getLParenLoc().getLocWithOffset(1),
      node->getRParenLoc().getLocWithOffset(-2)),
      SM, Result.Context->getLangOpts());
  
  auto tokenRange = Lexer::getAsCharRange(
      CharSourceRange::getTokenRange(node->getLParenLoc().getLocWithOffset(1),
      node->getRParenLoc().getLocWithOffset(-2)),
      SM, Result.Context->getLangOpts());
  
  llvm::outs() << "\n char range start: " << charRange.getBegin().printToString(SM)
                     << ",\n char range end: " << charRange.getEnd().printToString(SM) << "\n";

  llvm::outs() << "\n token range start: " << tokenRange.getBegin().printToString(SM)
                     << ",\n token range end: " << tokenRange.getEnd().printToString(SM) << "\n";
```

Tested on this snippet:
```
 int main() {
     int i = (int)7.777;
     return 0;
 }
```
The char range start/end will be 14/15, while the token range start/end will be 14/17, which is correct in this case:
```
   char range start: /UNN-C-Style-Checker/test/test2.cpp:2:14,
   char range end: /UNN-C-Style-Checker/test/test2.cpp:2:15
  
   token range start: /UNN-C-Style-Checker/test/test2.cpp:2:14,
   token range end: /UNN-C-Style-Checker/test/test2.cpp:2:17
```